Backend/mentii/class_ctrl.py

```diff
@@ -183,11 +186,144 @@ def getPublicClassList(dynamodb, email=None):
-    response.addError('Failed to get class list', 'A database error occured');
+    response.addError('Failed to get class list', 'A database error occured')
   else:
     res = classesTable.scan()
     for pclass in res.get('Items', []):
       if pclass['code'] not in classCodes and 'private' not in pclass and pclass.get('private') != True:
         classes.append(pclass)
     response.addToPayload('classes', classes)
   return response
+
+def removeStudent(dynamoDBInstance, jsonData, userRole=None):
+  response = ControllerResponse()
+  email = jsonData.get('email')
+  classCode = jsonData.get('classCode')
+  if g:
+    userRole = g.authenticatedUser['userRole']
+  if userRole != 'teacher' and userRole != 'admin':
+    response.addError('Role error', 'Only those with teacher privileges can remove students from classes')
+  elif email is None or classCode is None:
+    response.addError('Failed to remove student from class', 'Invalid data given')
+  else:
+    removeClassFromStudent(dynamoDBInstance, response, email, classCode)
+    if not response.hasErrors():
+      removeStudentFromClass(dynamoDBInstance, response, email, classCode)
+  return response
+
+def removeClassFromStudent(dynamoDBInstance, response, email, classCode):
+  usersTable = dbUtils.getTable('users', dynamoDBInstance)
+  if usersTable is None:
+    MentiiLogging.getLogger().error('Unable to get users table in removeStudentFromClass')
+    response.addError('Remove Student From Class Failed.', 'Unable to locate necessary table(s).')
+  else:
+    # check that user exists and user has that class
+    user = user_ctrl.getUserByEmail(email, dynamoDBInstance)
+    if user is None:
+      MentiiLogging.getLogger().error('Unable to get user by email in removeStudentFromClass')
+      response.addError('Failed to remove student from class', 'Unable to find user')
+    else:
+      classes = user['classCodes'] #set
+      if classCode not in classes:
+        response.addError('Failed to remove class from student', 'Class not found')
+      else:
+        classes.remove(classCode)
+        jsonData = {}
+        if len(classes) == 0:
+          #remove attribute
+          jsonData = {
+            'Key': {'email': email},
+            'UpdateExpression': 'REMOVE classCodes',
+            'ReturnValues' : 'UPDATED_NEW'
+          }
+        else:
+          #update attribute
+          jsonData = {
+            'Key': {'email': email},
+            'UpdateExpression': 'SET classCodes = :cc',
+            'ExpressionAttributeValues': { ':cc': classes },
+            'ReturnValues' : 'UPDATED_NEW'
+          }
+        res = dbUtils.updateItem(jsonData, usersTable)
+        if res is None:
+          MentiiLogging.getLogger().error('Unable to update classes in removeStudentFromClass')
+          response.addError('Failed to remove student from class', 'Unable to update user data')
+        else:
+          response.addToPayload('Class removal success', 'Removed class from student')
```

Comment:
It's not a big deal but remember that this is a key value pair that's being passed from the backed to the front end. Unless we do checks for a value 'Class removal success' on the fronted we shouldn't really be passing it.